controllers/topology/internal/extensions/patches/api/variables.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+const (
+	// BuiltinVariableName is the name of the builtin variable.
+	// NOTE: This also acts as a "namespace" preventing name collisions between
+	// user defined variables and builtin variables.
+	BuiltinVariableName = "builtin"
+)
+
+// BuiltinVariable represents builtin variables.
+type BuiltinVariable struct {
```

Comment:
Q: should this be in the variable package?

Reply:
It could be, just felt to me like the structure and which builtin variables we have is part of the API (like the actual calls). Thinking about external patch implementation and how they would access builtin variables.

I think it's also better to have a shared dependency between the variable and the inline package to the api package vs. that the inline patch package depends on the variable package (although it would be possible, there should be no cyclic dependencies)

Reply:
The idea of the API is to have a signature for things that will become RPC calls in future.

With the latest commit builtin variable are added as a single value of type object to the RPC call vs being many top level variables with nested names. I agree this makes consumption a little bit more complex, but I'm not sure sharing Golang types for builtin variable is the way to go to solve this problem, also because this makes an exception vs other variables (that are documented using open API schema) and go marshal/unmarshal does not fit into in two of the possible technologies we could leverage on for external patches (grpc/nor open API Schema); this also assumes extension implementers must import something from CAPI, which I'm not sure is something we want a constraint on, otherwise using other languages for developing exceptions became more complicated.

but I don't want to block on this given that the protocol for external patches is not defined yet and everything is internal.

Reply:
Sounds reasonable to me